### PR TITLE
Add import to pulumitest

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Adds an import function to pulumitest

The code was adapted from https://github.com/pulumi/pulumi-aws/pull/3859 through https://github.com/pulumi/pulumi-gcp/pull/1946

This is already used in GCP and AWS, I believe the azure provider needed it too at some point.

This is also needed for tests in the bridge: https://github.com/pulumi/pulumi-terraform-bridge/issues/2180

### CODE REVIEW

File: pulumitest/import_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestImport(t *testing.T) {
+	t.Parallel()
+	test := pulumitest.NewPulumiTest(t, "testdata/yaml_empty")
+
+	res := test.Import("random:index/randomString:RandomString", "str", "importedString", "")
+
+	require.Contains(t, res.Stdout, "type: random:RandomString")
```
Comment (thomas11) [inline pulumitest/import_test.go:16]: Seems a bit brittle to rely on particular output from pulumi. Can we assert on the stack contents instead?

**Review by thomas11 — APPROVED** [review]
LGTM with one comment.

Also, it's helpful when the PR description states the "why" of the change, intended usage etc.

File: pulumitest/import_test.go (hunk 1 above)
Comment (VenelinMartinov) [reply pulumitest/import_test.go:18]: This asserts on the generated code.

I've also added an assert for the state.

**Comment by VenelinMartinov** [comment]
Thanks, I've added notes on usage in AWS, GCP, Azure and the bridge.

**Comment by t0yv0** [comment]
@corymhall we were just talking this morning about the need for better import testing facilities. Looks like there is a start here, there's some examples here that write asserts against the generated program source code, but it does not yet help with tasks like taking the generated program and actually running it. Perhaps something that can be of interest to build on.

**Pull request merged**